### PR TITLE
Make `remove_stripe_titarenko` CPU-GPU agnostic

### PR DESCRIPTION
**Pull request by namannimmo10**
Made the name similar to TomoPy.

### CODE REVIEW

File: httomolib/prep/stripe.py [hunk 1]
```diff
@@ -109,12 +109,11 @@ def _rs_sort(sinogram, size, dim):
...
 
 
-## %%%%%%%%%%%%%%%%%%% remove_stripes_titarenko_cupy %%%%%%%%%%%%%%%%%%%%%%%  ##
 @nvtx.annotate()
-def remove_stripes_titarenko_cupy(
+def remove_stripe_ti(
     data: ndarray,
     beta: float = 0.1,
-) -> np.ndarray:
+) -> ndarray:
```
Comment (team-gpu) [inline httomolib/prep/stripe.py]: you can use `typings.Union` to give multiple possible types. like this:

```python
from typings import Union

def remove_stripe_ti(data: Union[cp.ndarray, np.ndarray], ...)
```

File: tests/test_prep/test_stripe.py [hunk 2]
```diff
@@ -6,26 +6,37 @@
...
     remove_stripe_based_sorting_cupy,
-    remove_stripes_titarenko_cupy,
+    remove_stripe_ti,
 )
 from numpy.testing import assert_allclose
 
 
 @cp.testing.gpu
-def test_stripe_removal_titarenko_cupy(data, flats, darks):
+def test_remove_stripe_ti_on_data(data, flats, darks):
```
Comment (team-gpu) [inline tests/test_prep/test_stripe.py:15]: I think now that it works with both numpy + cupy, why not add another test which generates random data and compares numpy to cupy output? With this: https://docs.cupy.dev/en/stable/reference/generated/cupy.testing.numpy_cupy_allclose.html

**Review by team-gpu — CHANGES_REQUESTED** [review]

File: httomolib/prep/stripe.py (hunk 1 above)
Comment (namannimmo10) [reply httomolib/prep/stripe.py]: Fair point. This can be helpful for readability, although python doesn't do strict type-checking.

Comment (team-gpu) [reply httomolib/prep/stripe.py]: IDEs typically underline if the types don't match, and it helps with syntax completion. We generally find strict type specifiers very helpful.

Comment (namannimmo10) [reply httomolib/prep/stripe.py]: Oh yes, strict type-checking is the best, helps in catching bugs easily.

**Pull request merged**